src/ERC20Whitelist.sol

```diff
@@ -1,53 +1,76 @@
 // SPDX-License-Identifier: MIT
 pragma solidity =0.7.6;
 
-import {IOracleWhitelist} from "./interfaces/IOracleWhitelist.sol";
-import {IApproveAndCallReceiver} from "./interfaces/IApproveAndCallReceiver.sol";
-import {IERC20Whitelist} from "./interfaces/IERC20Whitelist.sol";
-import {ERC20, ERC20Burnable} from "@openzeppelin/contracts/token/ERC20/ERC20Burnable.sol";
-import {ERC20Permit} from "@openzeppelin/contracts/drafts/ERC20Permit.sol";
-import {Ownable} from "@openzeppelin/contracts/access/Ownable.sol";
+import { IOracleWhitelist } from './interfaces/IOracleWhitelist.sol';
+import { IApproveAndCallReceiver } from './interfaces/IApproveAndCallReceiver.sol';
+import { IERC20Whitelist } from './interfaces/IERC20Whitelist.sol';
+import { ERC20, ERC20Burnable } from '@openzeppelin/contracts/token/ERC20/ERC20Burnable.sol';
+import { ERC20Permit } from '@openzeppelin/contracts/drafts/ERC20Permit.sol';
+import { Ownable } from '@openzeppelin/contracts/access/Ownable.sol';
 
 /**
  * @title Extended token contract with whitelist contract interactions
  * @notice During whitelist period, `_beforeTokenTransfer` function will call `checkWhitelist` function of whitelist contract
  * @notice If whitelist period is ended, owner will set whitelist contract address back to address(0) and tokens will be transferred freely
  */
-contract ERC20Whitelist is IERC20Whitelist, ERC20Burnable, ERC20Permit, Ownable {
+contract ERC20Whitelist is
+    IERC20Whitelist,
+    ERC20Burnable,
+    ERC20Permit,
+    Ownable
+{
     /// @dev whitelist contract address
     address public override whitelistContract;
 
     constructor(
-        address owner, 
-        string memory name, 
-        string memory symbol, 
+        address owner,
+        string memory name,
+        string memory symbol,
         uint256 _totalSupply
-    ) ERC20(name, symbol)
-        ERC20Permit(name) {
+    ) ERC20(name, symbol) ERC20Permit(name) {
         transferOwnership(owner);
         _mint(owner, _totalSupply);
     }
 
-    function setWhitelistContract(address _whitelistContract) external override onlyOwner {
+    function setWhitelistContract(
+        address _whitelistContract
+    ) external override onlyOwner {
         whitelistContract = _whitelistContract;
     }
 
-    function approveAndCall(address spender, uint256 amount, bytes calldata extraData) external returns (bool) {
+    function approveAndCall(
```

Comment:
Do we need this call for general tokens? Was used for Vultisig, but not sure if it should be applicable to any projects

Reply:
Token whitelist and token factory not in scope of this PR but noted for further discussion